### PR TITLE
Porting disable/enable functions using configuration

### PR DESCRIPTION
**Pull request by alrod**

### CODE REVIEW

File: src/Microsoft.Azure.WebJobs.Host/Extensions/IJobHostMetadataProvider.cs [hunk 1]
```diff
@@ -61,5 +61,12 @@ public interface IJobHostMetadataProvider
         /// </summary>
         /// <param name="output">The <see cref="TextWriter"/> to write the graph to.</param>
         void DebugDumpGraph(TextWriter output);
+
+        /// <summary>
+        /// Gets function metadata.
+        /// </summary>
+        /// <param name="functionName">Name of function to return metadata for.</param>
+        /// <returns></returns>
+        FunctionMetadata GetFunctionMetadata(string functionName);
```
Comment (mathewc) [inline src/Microsoft.Azure.WebJobs.Host/Extensions/IJobHostMetadataProvider.cs]: This unfortunately would be a breaking change to this public interface. Perhaps we don't need this though. The v1 functions host already has the settings logic for looking for these disable settings. That should be enough, meaning we don't have to flow this info via the metadata provider to the host.

File: src/Microsoft.Azure.WebJobs.Host/Listeners/HostListenerFactory.cs [hunk 2]
```diff
@@ -64,6 +58,12 @@ public async Task<IListener> CreateAsync(CancellationToken cancellationToken)
                     continue;
                 }
 
+                IListenerFactory listenerFactory = functionDefinition.ListenerFactory;
```
Comment (mathewc) [inline src/Microsoft.Azure.WebJobs.Host/Listeners/HostListenerFactory.cs:61]: Why this move?

Comment (alrod) [reply src/Microsoft.Azure.WebJobs.Host/Listeners/HostListenerFactory.cs:61]: we want to log the disabled function if there is no listener

File: src/Microsoft.Azure.WebJobs.Host/ConfigurationUtility.cs [hunk 3]
```diff
@@ -42,5 +42,20 @@ public static string GetConnectionFromConfigOrEnvironment(string connectionName)
 
             return Environment.GetEnvironmentVariable(connectionName) ?? configValue;
         }
+
+        public static bool IsSettingEnabled(string settingName)
```
Comment (mathewc) [inline src/Microsoft.Azure.WebJobs.Host/ConfigurationUtility.cs:46]: See Brett's PR https://github.com/Azure/azure-webjobs-sdk/pull/2513. This Type is going away

**Review by mathewc — APPROVED** [review]

**Pull request merged**